File: bike-shop/frontend/src/services/productService.js
import axiosClient from "./axiosConfig";

const productService = {
  getAll: (params) => axiosClient.get("/products", { params }),
  getById: (id) => axiosClient.get(`/products/${id}`),
  getFeatured: () => axiosClient.get("/products/featured"),
  getNewest: () => axiosClient.get("/products/newest"),
  create: (data) => axiosClient.post("/products", data),
  update: (id, data) => axiosClient.put(`/products/${id}`, data),
  delete: (id) => axiosClient.delete(`/products/${id}`),
};

export default productService;

File: bike-shop/bike-shop/backend/src/main/java/com/example/bike_shop/service/ProductService.java
package com.example.bike_shop.service;

import com.example.bike_shop.dto.ProductDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.math.BigDecimal;

public interface ProductService {
    Page<ProductDTO> searchProducts(String keyword, Long categoryId, Long brandId,
                                     BigDecimal minPrice, BigDecimal maxPrice,
                                     String sortBy, Pageable pageable);
    ProductDTO getById(Long id);
    ProductDTO create(ProductDTO dto);
    ProductDTO update(Long id, ProductDTO dto);
    void delete(Long id);
    java.util.List<ProductDTO> getFeatured();
    java.util.List<ProductDTO> getNewest();
    java.util.List<ProductDTO> getBestSellers();
}
